packages/fiori/src/FlexibleColumnLayout.ts

```diff
@@ -576,14 +619,18 @@ class FlexibleColumnLayout extends UI5Element {
 		this.separatorMovementSession = null;
 	}
 
-	saveUserDefinedColumnLayout(newLayout: FCLLayout, newColumnLayout: FlexibleColumnLayoutColumnLayout) {
-		const media = this.media as MEDIA.TABLET | MEDIA.DESKTOP;
-
-		this._userDefinedColumnLayouts[media][newLayout] = newColumnLayout;
+	saveUserDefinedColumnLayout(newLayout: FCLLayout, newColumnLayout: string[]) {
+		const media = this.media as MEDIA.TABLET | MEDIA.DESKTOP,
+			oldColumnLayout = this.getCustomColumnLayout(newLayout);
+		this.layoutsConfiguration[media] ??= {};
+		this.layoutsConfiguration[media]![newLayout] ??= { layout: newColumnLayout };
 		if (this.layout !== newLayout) {
 			this.layout = newLayout;
 			this.fireLayoutChange(true, false);
 		}
+		if (oldColumnLayout !== newColumnLayout) {
```

Comment:
Those two seems to be arrays, comparing them with "===" will compare them by reference and would result in false. Unless this is intentional, we may need to deep check if the fields have changed.

Reply:
thanks, corrected to compare the actual content